Optimize GMC e2e scripts

daisy-ycguo opened the pull request:
## Description

Optimize GMC e2e scripts to remove duplicated codes
Optimize GMC e2e workflow to reduce the latency

## Issues

closes issue #159

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

n/a

## Tests

CI tests

irisdingbj commented on .github/workflows/go-e2e.yaml:
```diff
@@ -9,6 +9,8 @@ on:
     paths:
       - microservices-connector/**
+      - manifests/ChatQnA/**
+      - manifests/DocSum/xeon/docsum*
```
shall we add the other two examples(codegen and codetrans) as well?

daisy-ycguo replied:
I think we've already added the test cases of codegen and codetrans. Yet GMC doesn't have any dependencies to manifest/CodeGen and manifest/CodeTrans. GMC only uses manifest/ChatQnA and manifests/DocSum/*/docsum_llm.yaml. So I just include these two folders in the workflow.

irisdingbj reviewed (commented):
Thanks @daisy-ycguo for taking care of GMC e2e test.   Besides GMC e2e, will will move all e2e tests in kind environments?  Any problems we keep using the original k8s env on the CI/CD machine?

daisy-ycguo commented:
> Thanks @daisy-ycguo for taking care of GMC e2e test. Besides GMC e2e,Will we move all e2e tests in kind environments? Any problems we keep using the original k8s env on the CI/CD machine?

We just moved GMC e2e to KIND environment because GMC controller has to be unique in K8s and it blocks parallel testing of workloads in GenAIExamples/GenAIInfra.

By using this workflow: [gmc-on-push.yaml](https://github.com/opea-project/GenAIInfra/blob/main/.github/workflows/gmc-on-push.yaml), GMC images are build and opea-system are installed and updated to K8s cluster when GMC source codes are changed.
